homeassistant/components/zodiac/sensor.py

```diff
@@ -165,42 +177,42 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
 class ZodiacSensor(SensorEntity):
     """Representation of a Zodiac sensor."""
 
-    def __init__(self):
+    def __init__(self) -> None:
         """Initialize the zodiac sensor."""
-        self._attrs = None
-        self._state = None
+        self._attrs: dict[str, str] = {}
+        self._state: str = ""
 
     @property
-    def unique_id(self):
+    def unique_id(self) -> str:
         """Return a unique ID."""
         return DOMAIN
 
     @property
-    def name(self):
+    def name(self) -> str:
         """Return the name of the entity."""
         return "Zodiac"
 
     @property
-    def device_class(self):
+    def device_class(self) -> str:
         """Return the device class of the entity."""
-        return "zodiac__sign"
+        return "zodiac_sign"
```

Comment:
The custom device classes for translation have a special format.
```suggestion
        return "zodiac__sign"
```